src/main/java/com/k3ntako/HTTPServer/ServerGenerator.java

```diff
@@ -0,0 +1,37 @@
+package com.k3ntako.HTTPServer;
+
+import com.k3ntako.HTTPServer.wrappers.ServerSocketWrapper;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.LinkedHashMap;
+
+public class ServerGenerator {
+  private FileIOInterface fileIO;
+  private YamlIOInterface yamlIOInterface;
+
+  public ServerGenerator(FileIOInterface fileIO, YamlIOInterface yamlIOInterface) {
+    this.yamlIOInterface = yamlIOInterface;
+    this.fileIO = fileIO;
+  }
+
+  public Server generate() throws Exception {
+    var config = this.getConfig();
+    var serverSocket = new ServerSocketWrapper((int) config.get("port"));
```

Comment:
What happens here if port exists but is an empty string?

Reply:
If the config is not an `int`, it would throw an error pointing to this line. I think we discussed that it wouldn't be necessary to do a type check.